app/src/main/java/org/mozilla/focus/searchsuggestions/ui/SearchOverlay.kt

```diff
@@ -0,0 +1,120 @@
+import mozilla.components.feature.awesomebar.provider.SearchSuggestionProvider
+import org.mozilla.focus.R
+import org.mozilla.focus.components
+import org.mozilla.focus.searchsuggestions.SearchSuggestionsViewModel
+import org.mozilla.focus.searchsuggestions.State
+import org.mozilla.focus.topsites.TopSitesOverlay
+import org.mozilla.focus.ui.theme.FocusTheme
+import org.mozilla.focus.ui.theme.focusColors
+
+@OptIn(DelicateCoroutinesApi::class)
+@Composable
+fun SearchOverlay(
+    viewModel: SearchSuggestionsViewModel,
+    defaultSearchEngineName: String,
+    onListScrolled: () -> Unit
+) {
+    val state = viewModel.state.observeAsState()
+    val query = viewModel.searchQuery.observeAsState()
+
+    when (state.value) {
+        is State.Disabled,
+        is State.NoSuggestionsAPI -> {
+            if (query.value.isNullOrEmpty()) {
+                TopSitesOverlay(modifier = Modifier.background(focusColors.surface))
+            }
+        }
+        is State.ReadyForSuggestions -> {
+            if (query.value.isNullOrEmpty()) {
+                TopSitesOverlay(modifier = Modifier.background(focusColors.surface))
+            } else {
+                SearchSuggestions(
+                    text = query.value ?: "",
+                    onSuggestionClicked = { suggestion ->
+                        viewModel.selectSearchSuggestion(
+                            suggestion.title!!,
+                            defaultSearchEngineName
+                        )
+                    },
+                    onAutoComplete = { suggestion ->
+                        val editSuggestion = suggestion.editSuggestion ?: return@SearchSuggestions
+                        viewModel.setAutocompleteSuggestion(editSuggestion)
+                    },
+                    onListScrolled = onListScrolled
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun SearchSuggestions(
+    text: String,
+    onSuggestionClicked: (AwesomeBar.Suggestion) -> Unit,
+    onAutoComplete: (AwesomeBar.Suggestion) -> Unit,
+    onListScrolled: () -> Unit
+) {
+    val context = LocalContext.current
+    val components = components
+
+    val icon = AppCompatResources.getDrawable(context, R.drawable.mozac_ic_search)?.toBitmap()
+    val provider = remember(context) {
+        SearchSuggestionProvider(
+            context,
+            components.store,
+            components.searchUseCases.newPrivateTabSearch,
+            components.client,
+            mode = SearchSuggestionProvider.Mode.MULTIPLE_SUGGESTIONS,
+            private = true,
+            showDescription = false,
+            icon = icon
+        )
+    }
+
+    val nestedScrollConnection = remember {
+        object : NestedScrollConnection {
+            override fun onPreScroll(available: Offset, source: NestedScrollSource): Offset {
+                onListScrolled.invoke()
+                return Offset.Zero
+            }
+        }
+    }
+
+    FocusTheme {
```

Comment:
This should not be needed here, as this should be in the fragment when we call `setContent` on a `ComposeView` or in a topmost composable in a complete Compose screen.